components/datatypes/consent-preferences.schema.json

```diff
@@ -0,0 +1,261 @@
+{
+  "meta:license": [
+    "Copyright 2020 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/datatypes/consents-and-preferences",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Privacy/Personalization/Marketing Preferences (Consent)",
+  "description": "This schema captures privacy, personalization and marketing preferences (consents).",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "definitions": {
+    "choice-value": {
+      "title": "Choice Value",
+      "description": "The value specified by the user as their preference or the basis of processing that precludes the need for user consent",
+      "type": "string",
+      "enum": ["y", "n", "p", "u", "LI", "CT", "CP", "VI", "PI"],
+      "meta:enum": {
+        "y": "Yes",
+        "n": "No",
+        "p": "Pending verification",
+        "u": "Unknown",
+        "LI": "Legitimate Interest",
+        "CT": "Contract",
+        "CP": "Compliance with a Legal Obligation",
+        "VI": "Vital Interest of the Individual",
+        "PI": "Public Interest"
```

Comment:
I originally had these split into two fields with the full strings. Both the combining and the abbreviations are to reduce the schema size as my original schema was deemed to be much too large.
The lower case, single letter options represent "consent" values, while the upper case, two-letter values represent basisOfProcessing values where user consent is not required. My understanding is that in the UI customers will see the full description for each value, so the shortened versions won't be a problem.
Customers told us that the terms in/out were confusing in some contexts, especially in the old consent schema: What does 
  "general_opt_out": "in"
mean? Are you opting in to opting out or are you not opting out?

Reply:
I think when you say "UI Customers" you're referring to the end user on our customer websites looking at, for example, the CMP UI. If so, I don't think the abbreviations are a problem for them either. I was thinking more about the Adobe customers that are trying to debug their consent integration and are inspecting network requests as they travel through Alloy (or the Mobile SDK), through the edge, and into their dataset, etc. I don't really know what/who is determining the threshold of "too large" though.

Regarding `in`/`out` being confusing in some contexts, that makes sense with `general_opt_out`. Is this new schema going to be used in confusing contexts like `general_opt_out`? I mainly ask because we discussed this problem at length with Karthik and others regarding those confusing contexts and came to a consensus that we would avoid the need for these confusing contexts in the web SDK and edge so that would could properly use `in`/`out` in a way that's understandable to customers. For example, you'll see this in the `defaultConsent` configuration of the SDK: https://experienceleague.adobe.com/docs/experience-platform/edge/fundamentals/configuring-the-sdk.html?lang=en#default-consent

Notice the two possible values for `defaultConsent` are `in` and `pending`. It would be confusing, IMO, to have them be `yes` and `pending` and I'd love for us to be consistent. There is no confusing concept of "configuring opt-out" in the web SDK or edge.

Reply:
I completely agree with aaronius evaluation here. We should be consistent and provide very clear options. I much prefer using the language "Opt-in" "Opt-out" (qualifies what in and out really mean).